test/watch-test.js

```diff
@@ -0,0 +1,290 @@
+/**
+ *
+ *  Web Starter Kit
+ *  Copyright 2016 Google Inc. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License
+ *
+ */
+
+/* eslint-env node, mocha */
+
+'use strict';
+
+require('chai').should();
+
+const fs = require('fs');
+const path = require('path');
+const ncp = require('ncp');
+const mkdirp = require('mkdirp');
+const rimraf = require('rimraf');
+const taskHelper = require('./helpers/task-helper');
+
+const VALID_TEST_FILES = path.join('test', 'data', 'valid-files');
+const INVALID_TEST_FILES = path.join('test', 'data', 'invalid-files');
+const VALID_TEST_FILES_2 = path.join('test', 'data', 'valid-files-2');
+const TEST_OUTPUT_PATH = path.join('test', 'output');
+const TEST_OUTPUT_SRC = path.join(TEST_OUTPUT_PATH, 'src');
+const TEST_OUTPUT_DEST = path.join(TEST_OUTPUT_PATH, 'build');
+
+let watcherTask;
+
+// Use rimraf over del because it seems to work more reliably on Windows.
+// Probably due to it's retries.
+const deleteFiles = path => {
+  return new Promise((resolve, reject) => {
+    rimraf(path, err => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      resolve();
+    });
+  })
+  .catch(err => {
+    console.log('deleteFiles() error ', err);
+    throw err;
+  });
+};
+
+const copyFiles = (from, to) => {
+  return new Promise((resolve, reject) => {
+    ncp(from, to, err => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      resolve();
+    });
+  })
+  .catch(err => {
+    console.log('copyFiles() error ', err);
+    throw err;
+  });
+};
+
+const validateOutput = () => {
+  // Get directories in build directory
+  const folders = fs.readdirSync(TEST_OUTPUT_DEST);
+  folders.forEach(folderName => {
+    // Check if the source directory lives, if it doesn't we can ignore
+    // the remaining output (i.e. old stuff that needs cleaning out)
+    try {
+      fs.lstatSync(path.join(TEST_OUTPUT_SRC, folderName));
+    } catch (err) {
+      // Path doesn't exist, we can ignore it
+      return;
+    }
+
+    const expectedOutputFileBuffer = fs.readFileSync(path.join(TEST_OUTPUT_SRC, folderName, 'output.json'));
+    const expectedOutput = JSON.parse(expectedOutputFileBuffer.toString());
+    expectedOutput.forEach(file => {
```

Comment:
Not unless we can't to care about deleting files as we go along (i.e. a watch kicks off a clean and then a build).

That opens issues with sass around cleaning CSS files up.